test/prepend.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+var throws = require('assert').throws;
+
+var errorEq = require('./utils').errorEq;
+var eq = require('./utils').eq;
+var S = require('..');
+
+
+describe('prepend', function() {
+
+  it('is a binary function', function() {
+    eq(typeof S.prepend, 'function');
+    eq(S.prepend.length, 2);
+  });
+
+  it('type checks its arguments', function() {
+    throws(function() { S.prepend('a', 'bc'); },
```

Comment:
What do you have in mind, Stefano? We test these failures:

```
prepend :: a -> Array a -> Array a
                ^^^^^^^
```

```
prepend :: a -> Array a -> Array a
           ^          ^
```

We don't test these failures:

```
prepend :: a -> Array a -> Array a
                           ^^^^^^^
```

```
prepend :: a -> Array a -> Array a
           ^                     ^
```

Assuming our function is implemented correctly, there's no way to test invalid return values. ;)